### PR TITLE
Modified TextAppearanceViewModelResourceManager to accept any attributes.

### PR DESCRIPTION
**Pull request by philippelamy1**
Modified TextAppearanceViewModelResourceManager to accept any attributes.

## Description

Instead of fixed attributes, an array of attributes is now in place. It can take any NSAttributedString.Key so we can now change a label any way we want to.
## Motivation and Context

We had to use another attribute that wasn't in place. 
Adding an array of attributes will be more flexible and more  permissive in the future.
## How Has This Been Tested?
- [X] All features has been added/updated in sample application ( /sample )
The feature has been tested in the sample

## Screenshots of sample application ( /sample ) (if appropriate):
Screen Shot 2020-11-17 at 4 17 09 PM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: sample/ios/iosApp/sample/SampleTextAppearanceResourceProvider.swift [hunk 1]
```diff
@@ -2,24 +2,28 @@ import Trikot_viewmodels
 import ViewModelsSample
 
 class SampleTextAppearanceResourceProvider: TextAppearanceViewModelResourceProvider {
-    func textAppearance(fromResource resource: TextAppearanceResource) -> TextAppearanceAttributes? {
+    func textAppearance(fromResource resource: TextAppearanceResource) -> TextAppearanceAttributes {
         if let sampleResource = resource as? SampleTextAppearanceResource {
             switch sampleResource {
             case .textAppearanceBold:
-                return TextAppearanceAttributes(font: .boldSystemFont(ofSize: 12), foregroundColor: .black)
+                return TextAppearanceAttributes(attributes: [.font: UIFont.boldSystemFont(ofSize: 12), .foregroundColor: UIColor.black])
```
Comment (antoinelamy) [inline sample/ios/iosApp/sample/SampleTextAppearanceResourceProvider.swift:9]: Why having an explicit `foregroundColor` instead of supplying `NSAttributedString.Key.foregroundColor` in the `attributes` parameter?

File: swift-extensions/TextAppearanceViewModelResourceManager.swift [hunk 2]
```diff
@@ -2,24 +2,20 @@ import UIKit
...
 public protocol TextAppearanceViewModelResourceProvider {
-    func textAppearance(fromResource resource: TextAppearanceResource) -> TextAppearanceAttributes?
+    func textAppearance(fromResource resource: TextAppearanceResource) -> TextAppearanceAttributes
 }
 
 class DefaultTextAppearanceViewModelResourceProvider: TextAppearanceViewModelResourceProvider {
-    func textAppearance(fromResource resource: TextAppearanceResource) -> TextAppearanceAttributes? {
-        nil
+    func textAppearance(fromResource resource: TextAppearanceResource) -> TextAppearanceAttributes {
+        TextAppearanceAttributes()
```
Comment (antoinelamy) [inline swift-extensions/TextAppearanceViewModelResourceManager.swift:18]: Why this change? `nil` would make more sense here IMO.

File: sample/common/src/commonMain/kotlin/com/trikot/viewModels/sample/resource/SampleTextAppearanceResource.kt [hunk 3]
```diff
@@ -7,5 +7,6 @@ enum class SampleTextAppearanceResource : TextAppearanceResource {
     TEXT_APPEARANCE_ITALIC,
     TEXT_APPEARANCE_COLORED,
     TEXT_APPEARANCE_GRAYED,
-    TEXT_APPEARANCE_HIGHLIGHTED
+    TEXT_APPEARANCE_HIGHLIGHTED,
+    TEXT_APPEARANCE_SUPERSCRIPT
```
Comment (antoinelamy) [inline sample/common/src/commonMain/kotlin/com/trikot/viewModels/sample/resource/SampleTextAppearanceResource.kt:11]: Seems unused

**Review by martingagnon — APPROVED** [review]
👍 After @antoinelamy comments

File: sample/common/src/commonMain/kotlin/com/trikot/viewModels/sample/resource/SampleTextAppearanceResource.kt (hunk 3 above)
Comment (antoinelamy) [reply sample/common/src/commonMain/kotlin/com/trikot/viewModels/sample/resource/SampleTextAppearanceResource.kt:11]: Nevermind, just realized this was sample code

File: sample/ios/iosApp/sample/SampleTextAppearanceResourceProvider.swift (hunk 1 above)
Comment (antoinelamy) [reply sample/ios/iosApp/sample/SampleTextAppearanceResourceProvider.swift:9]: A bit cross eyed this morning, `foregroundColor` is part of the array 🤦
